Lib/test/test_types.py

```diff
@@ -598,6 +598,18 @@ def test_method_descriptor_types(self):
         self.assertIsInstance(int.from_bytes, types.BuiltinMethodType)
         self.assertIsInstance(int.__new__, types.BuiltinMethodType)
 
+    def test_unsafe_new(self):
+        with self.assertRaises(TypeError) as cm:
+            object.__new__(dict)
+        message = 'object.__new__(dict) is not safe, use dict.__new__()'
+        self.assertEqual(str(cm.exception), message)
+
+    def test_unsafe_new_no_new(self):
+        with self.assertRaises(TypeError) as cm:
+            object.__new__(type(sys.flags))
+        message = 'object.__new__(sys.flags) is not safe'
+        self.assertEqual(str(cm.exception), message)
+
```

Comment:
Can you add a test for `tuple.__new__(type(sys.flags))`?

Reply:
That's easy, but I'm still not sure what is the right behavior here. I need time to read through 31506 to comprehend the problem.